src/states/elder/mod.rs

```diff
@@ -1637,6 +1623,25 @@ impl Approved for Elder {
         Ok(Transition::Stay)
     }
 
+    fn handle_neighbour_info_event(
+        &mut self,
+        elders_info: EldersInfo,
+        old_pfx: Prefix<XorName>,
+        neighbour_change: EldersChange,
+        outbox: &mut dyn EventBox,
+    ) -> Result<(), RoutingError> {
+        info!("{} - handle NeighbourInfo: {:?}.", self, elders_info);
+
+        if elders_info.prefix().is_extension_of(&old_pfx) {
+            self.finalise_split(outbox)?;
```

Comment:
We really do not want this happening, finalize split should only occur from a handle_section_info_event.
In practice this would never happen because we need to reach consensus on SectionInfo to send the NeighbourInfo, so they will always consensused after.

You can just delete that if and its content.